### DISCUSSION TITLE
Access blocked/ This app's request is invalid with Google Oauth

### DISCUSSION THREAD

**Discussion by stevenbaert** [open-webui/open-webui#4518] [Q&A]
Hi,

I'm running an Open-WebUI Docker image on my Synology NAS, which is behind a reverse proxy. It’s working fine, but I want to enable OAuth. I followed the guide at [Open-WebUI OAuth SSO Setup](https://docs.openwebui.com/tutorial/sso/), and I created a `client_id` and `client_secret` via the Google Console. However, I’m not entirely sure if I’ve configured the values correctly.

The Docker container is accessible locally at `http://10.0.6.77:8091`, and I’m using a reverse proxy to make it available externally via `https://chat.domain.dyndns.org`.

Here’s my current Google OAuth configuration:

**OAuth Settings:**
- **App Home Page:** `https://chat.domain.dyndns.org`
- **Authorized Domain:** `domain.dyndns.org`

**OAuth Credentials:**
- **Authorized JavaScript Origins:** `https://chat.domain.dyndns.org`
- **Authorized Redirect URIs:** `https://chat.domain.dyndns.org/open-webui/oauth/google/callback`

I then stopped the container, added the following environment variables (without rebuilding the image):

- `GOOGLE_CLIENT_ID`: My Google OAuth client ID
- `GOOGLE_CLIENT_SECRET`: My Google OAuth client secret
- `ENABLE_OAUTH_SIGNUP`: `true`

When I start the container and attempt to log in via Google, the Google login button appears, which suggests the setup is partially working. However, I’m getting an "Access blocked" error (screenshot attached).

I also have access to an official domain name and can create a subdomain if needed. Any advice on what I might be doing wrong?

Thanks for your help!

<img width="908" alt="Screenshot 2024-08-11 at 00 16 24" src="https://github.com/user-attachments/assets/c8121639-d09c-45e6-bb68-7cb11a05b886">

**Comment by stevenbaert** [comment]
Found issue: the redirect URI contained openwebui.

However now I can only allow new logins with oauth whereas I want to allow only logins I populated before. Also I want to only allow logins via oauth. 
Can that be done(?)

**Reply by justinh-rahb** [reply]
> However now I can only allow new logins with oauth whereas I want to allow only logins I populated before.

`OAUTH_MERGE_ACCOUNTS_BY_EMAIL=True`

> Also I want to only allow logins via oauth.

`ENABLE_LOGIN_FORM=False`

https://docs.openwebui.com/getting-started/env-configuration#oauth
